src/dataaggregator/aggregator/metricPoller.go

```diff
@@ -101,22 +109,23 @@ func (m *MetricPoller) doAggregate(appId string, metricType string, metrics []*m
 			sum += intValue
 		}
 	}
-
+	var avgAppMetric *appmetric.AppMetric
 	if count == 0 {
-		return &appmetric.AppMetric{
+		avgAppMetric = &appmetric.AppMetric{
```

Comment:
The AppMetric should only be constructed in one spot. just set the values in the if block.